### PR TITLE
meshdirectory: refactor to use ocm provider api

### PR DESCRIPTION
**Pull request by mirekys**
This moves most of the OCM provider functionality out of the `meshdirectory` package:

- Mesh Directory now leverages the `/cs3.ocm.provider.v1beta1.ProviderAPI/ListAllProviders` GRPC endpoint, that is already implemented by the `ocmproviderauthorizer` package
- Mentix driver, used for fetching OCM providers data from a Mentix exporter, has been moved to the `ocmproviderauthorizer` package
- Mesh Directory frontend now expects OCM provider data in the [ProviderInfo](https://github.com/cs3org/cs3apis/blob/master/cs3/ocm/provider/v1beta1/resources.proto#L77) struct format
- Mesh Directory package has been removed, only http service remains

Related to #870

### CODE REVIEW

**Comment by labkode** [comment]
@mirekys any update on this work?

**Comment by mirekys** [comment]
@labkode this is waiting for the Mentix exporter/GOCDB api to also use CS3 APIs structs (see here #870). The following code could then be removed from this pull, before it could be merged: https://github.com/cs3org/reva/pull/906/files#diff-ffed34085b7ccf28226b3573342026c2R115

**Comment by ishank011** [comment]
@Daniel-WWU-IT can you take a look? It'll be good to have all the services use a common mesh provider definition.

**Comment by Daniel-WWU-IT** [comment]
Just for my understanding: Mentix should export its provider data structured like the CS3API defines them? This surely can be done,  and makes a lot of sense. What you currently get from the /mentix endpoint is the "plain" data as it is stored internally. It's easy to extend that functionality and have it export the data in the CS3API structure format. From looking at their definitions, they should fit quite nicely, no need to extend them (at least for now).

Please confirm that I got this right and I'll work on it right away :)

**Comment by ishank011** [comment]
@Daniel-WWU-IT I think that should be it!

**Comment by mirekys** [comment]
@Daniel-WWU-IT yep, it should return a [ListAllProvidersResponse](https://github.com/cs3org/cs3apis/blob/master/cs3/ocm/provider/v1beta1/provider_api.proto#L106) formatted response, to be consistent with the OCM Provider api. You can have a look at the diff i referenced few comments ago to see, what the differences between that and current Mentix response are (not a ton of differences there :)).

File: pkg/ocm/provider/authorizer/mentix/mentix.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+			Description:  si.Description,
+			Organization: si.Organization,
+			Domain:       si.Domain,
+			Homepage:     si.Homepage,
+			Email:        si.Email,
+			Services:     services,
+		})
+	}
+
+	status := &rpcv1beta1.Status{
```
Comment (ishank011) [inline pkg/ocm/provider/authorizer/mentix/mentix.go:146]: You can just return providers here. We're not using status anywhere.

Comment (mirekys) [reply pkg/ocm/provider/authorizer/mentix/mentix.go:146]: True indeed, I will get rid of that

**Comment by Daniel-WWU-IT** [comment]
Created PR #955 to include a new CS3API exporter for Mentix. The data it exposes is just an array of `ProviderInfo` so you should be able to drop the conversion from the meshdirectory service.

**Comment by mirekys** [comment]
> @mirekys @ishank011 Created PR #955 to include a new CS3API exporter for Mentix. The data it exposes is just an array of `ProviderInfo` so you should be able to drop the conversion from the meshdirectory service.

thanks a lot @Daniel-WWU-IT, I'll update this PR right away

**Pull request merged**